components/TemplateGrid/TemplateGridContainer.vue

```diff
@@ -45,202 +45,142 @@ export default {
-                    item: {
-                        id: this.ghostIndex,
-                        column: this.lastColumn,
-                        row: this.lastRow,
-                        ghost: true,
-                    },
+        onDragOver(event) {
+            const { clientX, clientY } = event;
+            const { overRow, collisionRelation } = this.mousePosition;
+            if (this.onDragFirstItem(event)) return false;
+            const tmpCollisionRelation = this.getMouseOverProps(clientX, clientY);
+            const collidingItem = this.collisionsDetect(overRow);
+
+            if (collidingItem !== null && collisionRelation !== tmpCollisionRelation) {
+                this.setGhostItemPosition(this.getCollidingPosition(collidingItem));
+            } else if (collidingItem === null && overRow > this.maxRow) {
+                this.setGhostItemPosition({
+                    column: this.setAllowedColumn(),
+                    row: this.maxRow + this.positionBeetwenRows,
                 });
             }
+            event.preventDefault();
+            return true;
         },
-        getMouseOverProps(clientX, clientY) {
-            const elements = document.elementsFromPoint(clientX, clientY);
-            const shadowItem = elements.find(e => e.classList.contains('shadow-grid-item'));
-            const shadowItemId = shadowItem ? shadowItem.getAttribute('shadow-id') : null;
-            const overRow = shadowItemId ? Math.floor((shadowItemId - 1) / this.columns) : null;
-            const overColumn = shadowItemId ? (shadowItemId - 1) % this.columns : null;
-            const positionOverRow = shadowItemId
-                ? Math.floor(clientY - shadowItem.getBoundingClientRect().top)
-                : null;
-            return {
-                overColumn,
-                overRow,
-                positionOverRow,
-            };
+        onDragLeave(event) {
+            const { fromElement } = event;
+            if (fromElement && fromElement.classList.contains('blur')) {
+                this.removeGhostElement();
+            }
+            event.preventDefault();
         },
-        checkCollidingRelation(layerPositionY) {
-            const centerPosition = Math.floor(this.rowsHeight / 2);
-            return layerPositionY > centerPosition ? 'son' : 'brother';
+        onDrop(event) {
+            const { row, column } = this.ghostElement;
+            const { length: withoutGhostLength } = this.dataWithoutGhostElement;
+            this.removeGhostElement();
+            this.addItem({
+                id: this.draggedElement,
+                column,
+                row,
+                parent: this.setParentId(row, column),
+            });
+            if (withoutGhostLength >= this.rows - 2) {
+                this.$emit('setRowsCount', this.rows + 1);
+            }
+            this.$emit('rebuildGrid', this.draggedElement);
+            event.preventDefault();
+            return true;
```

Comment:
return true - not required / prevent default cancels event and returning true means that all went ok.